### ISSUE TITLE
error fetching eth1-deposits

### ISSUE THREAD

**Issue by imnisen** [gobitfly/eth2-beaconchain-explorer#288]
Hi, while I am trying to deploy and start the explorer following this [guide](https://github.com/gobitfly/eth2-beaconchain-explorer#getting-started) , I get the following error, any suggestions?

```
ubuntu@10-8-72-30:~/eth2-beaconchain-explorer$ ./bin/explorer --config ~/eth2-beaconchain-explorer/my-config.yaml
INFO[0000] config file path: /home/ubuntu/eth2-beaconchain-explorer/my-config.yaml
INFO[0000] database connection established
INFO[0000] gRPC connection to backend node established   module=rpc
INFO[0000] exported eth1-deposits                        blockHeight=3561403 depositsSaved=2 duration=31.931345ms fromBlock=3085928 module=exporter toBlock=3086928
ERRO[0005] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
INFO[0010] checking for new blocks/epochs to export      module=exporter
INFO[0010] Retrieving block at slot 495296               module=rpc
INFO[0010] caching assignements for epoch 15477          module=rpc
ERRO[0010] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0015] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0020] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0025] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0030] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0035] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0040] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0045] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0050] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0055] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0060] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0065] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0070] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0075] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0080] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0085] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0090] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0095] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0100] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0105] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0110] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0115] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0120] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0125] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0130] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0135] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0140] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
ERRO[0145] error fetching eth1-deposits                  error="error getting eth1-blocks: json: cannot unmarshal object into Go value of type []rpc.jsonrpcMessage" module=exporter
```

Where could I make the wrong things? Thanks!

**Comment by imnisen** [comment]
Does it only support the eth1 endpoint  with infrua? Could I use geth rpc? I find that exporter use grpc.

```
rpcClient, err := gethRPC.Dial(utils.Config.Indexer.Eth1Endpoint)
```

**Comment by guybrush** [comment]
I think this is because our eth1deposit-importer is currently only working with infura.

Will look into this, we definitely want it to work with a regular eth1-node!

**Comment by VieYang** [comment]
I have the same error. I use `geth --http` as the `indexer.eth1Endpoint`.

**Comment by guybrush** [comment]
@VieYang sorry! totally missed your commit :(

This is also fixed in 0e13c04b9

**Comment by bsean95531-prog** [comment]
Recovering what possible back in publicorg
